### PR TITLE
Feature/65 dex show and harwest the referral pending rewards

### PR DESCRIPTION
**Pull request by nicholas-aternum**

### CODE REVIEW

File: apps/dex/src/components/portfolio/ReferralCard.tsx [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
 
-export function ReferralCard() {
+interface ReferralCardProps {
+  pendingRewards: TokenReward[];
+  isPending: boolean;
+  refetchPendingRewards: () => Promise<QueryObserverResult<readonly [bigint, bigint, bigint, bigint], unknown>>;
+  writeHarvest: () => void;
+}
+
+export const ReferralCard: React.FC<ReferralCardProps> = ({
```
Comment (fredev) [inline apps/dex/src/components/portfolio/ReferralCard.tsx:17]: use functions instead of const

File: apps/dex/src/hooks/useReferralRewards.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		abi: referralsRewardsAbi,
+		address: referralRewardAddress as EthereumAddress,
+		functionName: 'rewards',
+		args: [address as EthereumAddress],
+	});
+
+  useEffect(() => {
+    if (data && data.length > 0) {
+      const mappedRewards = tokens.map((token, index) => {
+        if (token.symbol === 'LDT' || token.symbol.includes('TB')) {
```
Comment (fredev) [inline apps/dex/src/hooks/useReferralRewards.ts:32]: tokens can only be LDT and TB, this if is not needed

File: apps/dex/src/hooks/useReferralRewards.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        functionName: 'harvest',
+        args: [],
+      });
+    }
+  };
+
+  const confirmed = useWatchTransaction(writeData);
+
+  const refetchRewardsAndHarvest = async () => {
+    await refetch();
```
Comment (fredev) [inline apps/dex/src/hooks/useReferralRewards.ts:62]: why refetch before harwest?

File: apps/dex/src/hooks/useReferralRewards.ts (hunk 2 above)
Comment (nicholas-aternum) [reply apps/dex/src/hooks/useReferralRewards.ts:32]: I improve a mapping, but data is: [bigint, bigint, bigint, bigint] and not included specific token information. This approach allows us to accurately map and display the rewards with the corresponding token information.

File: apps/dex/src/hooks/useReferralRewards.ts (hunk 3 above)
Comment (nicholas-aternum) [reply apps/dex/src/hooks/useReferralRewards.ts]: Ok, removed.

**Review by fredev — APPROVED** [review]

**Pull request merged**